libs/langchain/langchain/agents/output_parsers/react_json_single_input.py

```diff
@@ -0,0 +1,76 @@
+import json
+import re
+from typing import Union
+
+from langchain.agents.agent import AgentOutputParser
+from langchain.agents.chat.prompt import FORMAT_INSTRUCTIONS
+from langchain.schema import AgentAction, AgentFinish, OutputParserException
+
+FINAL_ANSWER_ACTION = "Final Answer:"
+
+
+class ReActJsonSingleInputOutputParser(AgentOutputParser):
+    """Parses ReAct-style LLM calls that have a single tool input in json format.
+
+    Expects output to be in one of two formats.
+
+    If the output signals that an action should be taken,
+    should be in the below format. This will result in an AgentAction
+    being returned.
+
+    ```
+    Thought: agent thought here
+    Action:
+    ```
+    {
+        "action": "search",
+        "action_input": "what is the temperature in SF"
+    }
+    ```
+    ```
+
+    If the output signals that a final answer should be given,
+    should be in the below format. This will result in an AgentFinish
+    being returned.
+
+    ```
+    Thought: agent thought here
+    Final Answer: The temperature is 100 degrees
+    ```
+
+    """
+
+    pattern = re.compile(r"^.*?`{3}(?:json)?\n(.*?)`{3}.*?$", re.DOTALL)
```

Comment:
can we just have one pattern constant somewhere that all the json markdown things use